src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -191,7 +194,39 @@ public CallOutput EstimateGas(BlockHeader header, Transaction tx, CancellationTo
                 estimateGasTracer.WithCancellation(cancellationToken));
             
             long estimate = estimateGasTracer.CalculateEstimate(tx);
-            return new CallOutput {Error = tryCallResult.Success ? estimateGasTracer.Error : tryCallResult.Error, GasSpent = estimate, InputError = !tryCallResult.Success};
+            
+            return new CallOutput 
+            {
+                Error = tryCallResult.Success ? estimateGasTracer.Error : tryCallResult.Error, 
+                GasSpent = estimate, 
+                InputError = !tryCallResult.Success
+            };
+        }
+
+        public CallOutput CreateAccessList(BlockHeader header, Transaction tx, CancellationToken cancellationToken, bool optimize)
+        {
+            CallOutputTracer callOutputTracer = new();
+            AccessTxTracer accessTxTracer = optimize 
+                ? new(tx.SenderAddress, 
+                    tx.GetRecipient(tx.IsContractCreation ? _stateReader.GetNonce(header.StateRoot, tx.SenderAddress) : 0)) 
+                : new();
+
+            (bool Success, string Error) tryCallResult = TryCallAndRestore(header, header.Number, header.Timestamp, tx,
+                new CancellationTxTracer(new CompositeTxTracer(callOutputTracer, accessTxTracer), cancellationToken)
+                {
+                    IsTracingActions = true, 
+                    IsTracingOpLevelStorage = true,
+                    IsTracingInstructions = true, // a little bit costly but almost all are simple calls
+                    IsTracingRefunds = true
+                });
+            return new CallOutput
+            {
```

Comment:
is it violating DRY with other calls?

Reply:
not  really, its constructed somewhat differently each time.